ecr/base.go

```diff
@@ -50,36 +51,64 @@ type ecrAPI interface {
 	PutImageWithContext(aws.Context, *ecr.PutImageInput, ...request.Option) (*ecr.PutImageOutput, error)
 }
 
-func (b *ecrBase) getManifest(ctx context.Context) (*ecr.Image, error) {
-	imageIdentifier := b.ecrSpec.ImageID()
-	log.G(ctx).WithField("imageIdentifier", imageIdentifier).Debug("ecr.base.manifest: getting image")
+// getImageByDescriptor retrieves an image from ECR for a given OCI descriptor.
+func (b *ecrBase) getImageByDescriptor(ctx context.Context, desc ocispec.Descriptor) (*ecr.Image, error) {
+	input := ecr.BatchGetImageInput{
+		ImageIds: []*ecr.ImageIdentifier{
+			&ecr.ImageIdentifier{ImageDigest: aws.String(desc.Digest.String())},
+		},
+	}
+	if desc.MediaType != "" {
+		input.AcceptedMediaTypes = []*string{aws.String(desc.MediaType)}
+
+	}
 
-	batchGetImageInput := &ecr.BatchGetImageInput{
-		RegistryId:     aws.String(b.ecrSpec.Registry()),
-		RepositoryName: aws.String(b.ecrSpec.Repository),
-		ImageIds:       []*ecr.ImageIdentifier{imageIdentifier},
+	imgs, err := b.runGetImage(ctx, input)
```

Comment:
After changing the return type of `b.runGetImage` (as suggested below) you can simplify this as follows:

```go
return b.runGetImage(ctx, input)
```

Reply:
Thanks, there were some evolutionary stages and the single image case ended up being the only one necessary. It's been reworked and should be reflected in the latest push.